hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotDefragService.java

```diff
@@ -0,0 +1,1138 @@
+                cfName, currentSnapshot.getName(), e.getMessage(), e);
+          }
+
+          // Ingest SST file into target database if there were changes
+          if (cfChanges > 0 && sstFile.exists() && sstFile.length() > 0) {
+            try {
+              targetTable.loadFromFile(sstFile);
+              LOG.info("Successfully ingested SST file for column family {}: {} changes",
+                  cfName, cfChanges);
+            } catch (Exception e) {
+              LOG.error("Failed to ingest SST file for column family {}: {}", cfName, e.getMessage(), e);
+            }
+          } else {
+            LOG.debug("No changes to ingest for column family {}", cfName);
+          }
+
+          // Clean up SST file after ingestion
+          try {
+            if (sstFile.exists()) {
+              Files.delete(sstFile.toPath());
+              LOG.debug("Cleaned up SST file: {}", sstFile.getAbsolutePath());
+            }
+          } catch (IOException e) {
+            LOG.warn("Failed to clean up SST file: {}", sstFile.getAbsolutePath(), e);
+          }
+
+          totalChanges += cfChanges;
+          LOG.debug("Applied {} incremental changes for column family: {}", cfChanges, cfName);
+        }
+
+
+//        String lastToken = new String(diffList.get(diffList.size() - 1).getSourcePath(), StandardCharsets.UTF_8);
+        nextToken += diffList.size();
+        LOG.debug("Retrieving next page of snapshot diff with token: {}", nextToken);
+        diffResponse = ozoneManager.snapshotDiff(
+            currentSnapshot.getVolumeName(),
+            currentSnapshot.getBucketName(),
+            previousSnapshot.getName(),
+            currentSnapshot.getName(),
+            String.valueOf(nextToken), // token
+            0,    // pageSize
+            false, // forceFullDiff
+            false  // disableNativeDiff
+        );
+
+        if (diffResponse.getJobStatus() != SnapshotDiffResponse.JobStatus.DONE) {
+          throw new RocksDatabaseException("Expecting DONE but got unexpected snapshot diff status: " +
+              diffResponse.getJobStatus() + ", reason: " + diffResponse.getReason());
+        }
+
+        LOG.info("Retrieved next page of snapshot diff, size: {}",
+            diffResponse.getSnapshotDiffReport().getDiffList().size());
+        diffReport = diffResponse.getSnapshotDiffReport();
+      }
+
+      // Clean up temporary directory
+      try {
+        Files.deleteIfExists(Paths.get(tempSstDir));
+        LOG.debug("Cleaned up temporary SST directory: {}", tempSstDir);
+      } catch (IOException e) {
+        LOG.warn("Failed to clean up temporary SST directory: {}", tempSstDir, e);
+      }
+
+    } catch (IOException e) {
+      throw new RocksDatabaseException("Failed to create temporary SST directory: " + tempSstDir, e);
+    }
+
+    LOG.info("Applied {} total incremental changes using snapshotDiff approach", totalChanges);
```

Comment:
BTW we should truncate other tables like deletedTable, deletedDirectoryTable, snapshotRenamedTable, bucketTable, volumeTable etc. and directly bulk import the tables into the defragged snapshot DB.
Here we can use: dumpToFile and loadFromFile which would just dump the table into an sstFile which can be directly loaded. This would be much more optimal.
https://github.com/apache/ozone/blob/ae92a189d37b98071b78b075163757c7b4b79aec/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java#L469-L477